File: Examples/DotNET/CSharp/Document/ConvertToOtherFormatWithOutStorage.cs
using System;
using System.Collections.Generic;
using Com.Aspose.Words.Api;
using Com.Aspose.Words.Model;
using Com.Aspose.Storage.Api;

namespace Document
{
    class ConvertToOtherFormatWithOutStorage
    {
        public static void Run()
        {
            // ExStart:1
            WordsApi wordsApi = new WordsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            string format = "pdf";
            string outPath = null;
            string replaceResourcesHostTo = "new_pdf.pdf";
            byte[] file = System.IO.File.ReadAllBytes( Common.GetDataDir() + "test_uploadfile.docx");

            try
            {
                // Invoke Aspose.Words Cloud SDK API
                ResponseMessage apiResponse = wordsApi.PutConvertDocument(format, outPath, replaceResourcesHostTo, file);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    System.Diagnostics.Debug.WriteLine("response:" + apiResponse.ResponseStream);
                    
                    System.IO.File.WriteAllBytes("\\temp\\" + replaceResourcesHostTo, apiResponse.ResponseStream);
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/Document/ConvertToOtherFormatWithOutStorage.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeWordsCloud::WordsApi;
use AsposeWordsCloud::ApiClient;
use AsposeWordsCloud::Configuration;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeWordsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeWordsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeWordsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Words API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $wordsApi = AsposeWordsCloud::WordsApi->new();

# Set input file name
my $filename = "SampleWordDocument";
my $name = $filename . ".docx";
my $format = "pdf";

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
# Invoke Aspose.Words Cloud SDK API to convert words document to required format       
$response = $wordsApi->GetDocumentWithFormat(name=> $name, format=>$format);

if($response->{'Status'} eq 'OK'){
	# Download output document from response 
	my $output_file = $out_path. $filename . '.' . $format;	
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1